File: src-tauri/src/models/udp_broadcast_bridge.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UdpBroadcastBridgeConfig {
    pub id: Option<String>,
    pub listen_host: String,
    pub listen_port: u16,
    pub forward_targets: Vec<String>,
    pub label: Option<String>,
    pub game_id: Option<String>,
    pub allow_broadcast: Option<bool>,
    pub duplicate_ttl_ms: Option<u64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UdpBroadcastBridgeStatus {
    pub id: String,
    pub running: bool,
    pub listen: String,
    pub forward_targets: Vec<String>,
    pub received_packets: u64,
    pub forwarded_packets: u64,
    pub dropped_packets: u64,
    pub bytes_in: u64,
    pub bytes_out: u64,
    pub last_error: Option<String>,
    pub logs: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UdpBroadcastBridgeSelfTestReport {
    pub ok: bool,
    pub listen: String,
    pub forward_targets: Vec<String>,
    pub sent: String,
    pub received: String,
    pub received_packets: u64,
    pub forwarded_packets: u64,
    pub dropped_packets: u64,
    pub bytes_in: u64,
    pub bytes_out: u64,
    pub notes: Vec<String>,
    pub status: UdpBroadcastBridgeStatus,
}

File: src/types/udpBroadcastBridge.ts
export interface UdpBroadcastBridgeConfig {
  id?: string | null;
  listen_host: string;
  listen_port: number;
  forward_targets: string[];
  label?: string | null;
  game_id?: string | null;
  allow_broadcast?: boolean | null;
  duplicate_ttl_ms?: number | null;
}

export interface UdpBroadcastBridgeStatus {
  id: string;
  running: boolean;
  listen: string;
  forward_targets: string[];
  received_packets: number;
  forwarded_packets: number;
  dropped_packets: number;
  bytes_in: number;
  bytes_out: number;
  last_error?: string | null;
  logs: string[];
}

export interface UdpBroadcastBridgeSelfTestReport {
  ok: boolean;
  listen: string;
  forward_targets: string[];
  sent: string;
  received: string;
  received_packets: number;
  forwarded_packets: number;
  dropped_packets: number;
  bytes_in: number;
  bytes_out: number;
  notes: string[];
  status: UdpBroadcastBridgeStatus;
}
